File: scalaz-client/src/main/scala/io/swagger/client/api/FieldsEdgeModuleObject.scala
package io.swagger.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime
import FieldsEdgeModuleObject._

case class FieldsEdgeModuleObject (
  name: String,
fields: Option[List[FieldObjectFields]],
custom: Option[Boolean])

object FieldsEdgeModuleObject {
  import DateTimeCodecs._

  implicit val FieldsEdgeModuleObjectCodecJson: CodecJson[FieldsEdgeModuleObject] = CodecJson.derive[FieldsEdgeModuleObject]
  implicit val FieldsEdgeModuleObjectDecoder: EntityDecoder[FieldsEdgeModuleObject] = jsonOf[FieldsEdgeModuleObject]
  implicit val FieldsEdgeModuleObjectEncoder: EntityEncoder[FieldsEdgeModuleObject] = jsonEncoderOf[FieldsEdgeModuleObject]
}

File: dart-client/lib/model/fields_edge_module_object.dart
part of swagger.api;

class FieldsEdgeModuleObject {
  
  String name = null;
  

  List<FieldObjectFields> fields = [];
  

  bool custom = null;
  
  FieldsEdgeModuleObject();

  @override
  String toString() {
    return 'FieldsEdgeModuleObject[name=$name, fields=$fields, custom=$custom, ]';
  }

  FieldsEdgeModuleObject.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    name =
        json['name']
    ;
    fields =
      FieldObjectFields.listFromJson(json['fields'])
;
    custom =
        json['custom']
    ;
  }

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'fields': fields,
      'custom': custom
     };
  }

  static List<FieldsEdgeModuleObject> listFromJson(List<dynamic> json) {
    return json == null ? new List<FieldsEdgeModuleObject>() : json.map((value) => new FieldsEdgeModuleObject.fromJson(value)).toList();
  }

  static Map<String, FieldsEdgeModuleObject> mapFromJson(Map<String, Map<String, dynamic>> json) {
    var map = new Map<String, FieldsEdgeModuleObject>();
    if (json != null && json.length > 0) {
      json.forEach((String key, Map<String, dynamic> value) => map[key] = new FieldsEdgeModuleObject.fromJson(value));
    }
    return map;
  }
}

